hkmc2/shared/src/main/scala/hkmc2/codegen/HandlerLowering.scala

```diff
@@ -0,0 +1,561 @@
+package hkmc2
+package codegen
+
+import mlscript.utils.*, shorthands.*
+import utils.*
+
+import hkmc2.Message.MessageContext
+
+import semantics.Elaborator.State
+
+import syntax.{Literal, Tree, ParamBind}
+import semantics.*
+import scala.annotation.tailrec
+import hkmc2.semantics.Elaborator.ctx
+
+import hkmc2.utils.SymbolSubst
+
+object HandlerLowering:
+
+  private val pcIdent: Tree.Ident = Tree.Ident("pc")
+  private val nextIdent: Tree.Ident = Tree.Ident("next")
+  private val tailIdent: Tree.Ident = Tree.Ident("tail")
+
+  extension (k: Block => Block)
```

Comment:
These look like pretty general helpers. Maybe they belong in a more general file, perhaps even `Block.scala`.

Reply:
Same remark for `extension (p: Path)`.

Reply:
Yes I agree, I actually made the exact comment on the stack-safety branch. I'll do that once I address some other things.